src/redisearch_rs/buffer/src/lib.rs

```diff
@@ -121,3 +121,16 @@ impl Buffer {
         self.0.offset += n;
     }
 }
+
+impl std::fmt::Debug for Buffer {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut debug = f.debug_struct("Buffer");
+        let debug = debug
+            .field("len", &self.len())
+            .field("capacity", &self.capacity());
+        // We don't want to accidentally output huge or sensitive data in production code.
+        #[cfg(debug_assertions)]
+        let debug = debug.field("data", &self.as_slice());
+        debug.finish()
```

Comment:
This syntax means we assert if we reach here from a release build?

Reply:
No, see it like a `#ifdef` on this single line.

That means the data field will only be present in debug builds but not in release ones.

Reply:
The rationale for this is to prevent accidentally printing huge and/or sensitive data in release builds, see https://github.com/RediSearch/RediSearch/pull/6378#discussion_r2169128764